internal/resources/azure/storage_table.go

```diff
@@ -66,40 +75,17 @@ func (r *StorageTable) BuildResource() *schema.Resource {
 	}
 }
 
-func (r *StorageTable) isAccountKindSupported() bool {
-	return r.isStorageV1() || r.isStorageV2()
-}
-
 func (r *StorageTable) isReplicationTypeSupported() bool {
-	var validReplicationTypes []string
+	validReplicationTypes := []string{"LRS", "ZRS", "GRS", "RA-GRS", "GZRS", "RA-GZRS"}
 
-	switch {
-	case r.isStorageV1():
-		validReplicationTypes = []string{"LRS", "GRS", "RA-GRS"}
-	case r.isStorageV2():
-		validReplicationTypes = []string{"LRS", "ZRS", "GRS", "RA-GRS", "GZRS", "RA-GZRS"}
-	}
-
-	if validReplicationTypes != nil {
-		return contains(validReplicationTypes, strings.ToUpper(r.AccountReplicationType))
-	}
-
-	return true
+	return contains(validReplicationTypes, strings.ToUpper(r.AccountReplicationType))
 }
 
-func (r *StorageTable) isStorageV1() bool {
-	return strings.EqualFold(r.AccountKind, "storage")
-}
-
-func (r *StorageTable) isStorageV2() bool {
-	return strings.EqualFold(r.AccountKind, "storagev2")
-}
-
-func (r *StorageTable) productName() string {
-	if r.isStorageV1() {
-		return "Tables"
+func (r *StorageTable) lookupPrefix() string {
+	if r.HasCustomerManagedKey {
+		return "Account Encrypted"
```

Comment:
If there's a customer managed key we add a `Account Encrypted` prefix to the product lookups since they are all prefixed with this in the database. Otherwise we use `Standard`